### PR TITLE
Small content additions 

### PR DESCRIPTION
**Pull request by Booraz149**
**Cube-pistol**
Basically a rail pistol with "cube" enemy theme.
Infinite ammo, costs matter to shoot like Cube-gun.
25 damage per shot.
Can only drop from cubes; 2x drop chance compared to Cube-gun (and only can drop when Cube-gun doesn't)
Slot 1.

**Main menu change**
Set height of .set_block:focus-within to auto, max-height to 150px;
Set overflow-y to auto so it can automatically create a vertical scroll bar when needed.

**New helmet variations**
Since main menu had some interface changes, they allow more helmets.
4 new helmets, sprites by butorinoks77.

**New mob/enemy: Slug**
They only currently spawn from destroying sdBlock (chance somewhere between Octopus and Quickie)
They move around the map but do not attack if their health is max, they're not hostile on spawn.
If they have less than max health, they chase and attack nearest of the following (behaviour similar to Quickie):
sdBlock
sdVirus
sdAsp
sdQuickie
sdCharacter

Slugs regenerate 3hp per attack they do, even from blocks.
Slugs' attacks deal 30 damage.
Slugs have 500 health.
Once they regenerate back their behaviour goes back to passive.
Sprites could use an improvement in my opinion.

**New upgrade for characters: matter regeneration**
New upgrade, allowing players to regenerate matter on their own.
Accesible via wrench (sdShop)
Max 5 levels, 200 matter per level ( Change it however you'd like)
Matter regeneration is always 1 matter per second, however levels determine max value that character can regenerate
for example, level 1 allows you only to regenerate if below 20 matter, while level 5 allows up to 100 matter. ( Change it however you'd like)

### CODE REVIEW

**Comment by Booraz149** [comment]
Closed currently since I accidentally deleted server.bat
The pull request became broken (due to my errorrs)

**Comment by Eric-Gurt** [comment]
I think I'm still able to cancel portions of pull request in cases like this, but can't reopen when it is fully closed.

Also, due to simplistic logic of sync SD2D has - all entity properties that are not prefixed with _ will be sent to clients in each frame. Thus it could make sense to have variable 'matter_upg_max' to be prefixed in all files like this: '_matter_upg_max'. Could save some bytes in sync logic especially since property name is sent too.

SD2D so far does also handle matter as floating point numbers so there might be no need to round them. Also world logic frame can be varying on both server and client event though it is currently capped at 1 at server-side - for this reason adding anything or subtracting anything in any entities' logic should be multiplied by GSPEED which will be a relative time for frame. Usually something like `x += 1 / 30 * GSPEED;` will cause x to change from 0 to 1 within 1 second (30 = 1 second there).

Also, SD2D does round most of number properties when sending - for this reason matter appears rounded on client-side but it really isn't.

For matter increase I usually would do something like this.matter = Math.min( this.matter + GSPEED * 0.1, this.matter_max );
So in worst case scenario when player would try to buy something that costs this.matter_max  - he would have enough matter.

TL;DR: In case if you want any questions regarding how it can be best to use GSPEED - I think I can either fix it myself or describe why and how it would work in specific cases. Like there is also common method I use which is sdWorld.MorphWithTimeScale which is for non-linear changes but is able to handle soft changes that properly work with GSPEED value.
